src/capture-screenshot.js

```diff
@@ -0,0 +1,155 @@
+const puppeteer = require('puppeteer');
+const {performance} = require('perf_hooks');
+
+const devicePixelRatio = 1.0;
+const maxTimeInSec = 30;
+
+const timeDelta = (start, end) => {
+  return ((end - start) / 1000).toPrecision(3);
+}
+
+const htmlTemplate = (options) => {
+  const {width, height, inputPath, libPort, backgroundColor} = options;
+  return `
+    <html>
+      <head>
+        <meta name="viewport" content="width=device-width, initial-scale="${devicePixelRatio}">
+        <script type="module"
+          src="http://localhost:${libPort}/model-viewer.js">
+        </script>
+        <style>
+          body {
+            margin: 0;
+          }
+          model-viewer {
+            --progress-bar-color: transparent;
+            width: ${width};
+            height: ${height};
+          }
+        </style>
+      </head>
+      <body>
+        <model-viewer
+          style="background-color: ${backgroundColor};"
+          background-color=""
+          id="snapshot-viewer"
+          interaction-prompt="none"
+          src="${inputPath}" />
+      </body>
+    </html>
+  `
+}
+
+const captureScreenshot = async function (options) {
+  const browserT0 = performance.now();
```

Comment:
These could probably be cleaned up by doing something like:
`performance.mark('browserStartup');`
Then later:
`performance.measure('browserStartup').duration`

Reply:
They ended up removing returning the PerformanceEntry from the return of measure in Node 10 and prefer to use PerformanceObserver which makes the code a bit more involved.